dashboard/app/models/programming_environment.rb

```diff
@@ -73,7 +81,7 @@ def summarize_for_edit
     }
   end
 
-  def categories
+  def category_options
     programming_expressions.pluck(:category).uniq
```

Comment:
Why do you still need to use programming expressions to get category here?

Reply:
Oh I meant to add a comment here -- thanks for asking! tldr: this will get changed in a future PR.

This method is only used to provide the category options for the programming expression edit page, but ProgrammingEnvironmentCategory and ProgrammingExpression don't have an association set up yet.  When I make that change, I'll remove this method and use ProgrammingExpressionCategory as the options on that page. I originally removed this method but it broke the programming expression edit page and this PR had already gotten big enough, so I decided to add it back and address it later.